src/Umbraco.Core/Persistence/Repositories/Implement/DocumentRepository.cs

```diff
@@ -869,6 +905,50 @@ protected override string GetBaseWhereClause()
 
         #endregion
 
+        #region Schedule
+
+        /// <inheritdoc />
+        public IEnumerable<IContent> GetContentForRelease(DateTime date)
+        {
+            var action = ContentScheduleChange.Start.ToString();
+
+            // fixme/review - code would blow if more than 2000 items
+            // fixme/review - isn't this simpler?
```

Comment:
@zpqrtbnk The logic that checks for the !x.Published i think is incorrect (i know i had it there but was part of the logic from v7). The reason why this doesn't work is because the item will not be found for release even though the schedule dates are correct which means the schedule will remain in the database. It gets cleared with the logic in `PerformScheduledPublish` based on the `IContent` returned there.  I'm going to update this to remove the checks for Published and Trashed so both `GetContentForRelease` and `GetContentForExpiration` will return anything that a schedule matches for regardless of whether it is already published or in the trash. Then the caller can deal with each result appropriately.

Reply:
Hrm, actually i guess there's an argument for not returning published/trashed too but that means that in the `PerformScheduledPublish` we clear the schedule via SQL for all past dates so will need to make another method on the repo.

Reply:
Doh, sorry for all these messages. Ok, so the other problem with the x.Published check is that this is a problem for Drafts and Variants. This will return no results for a content item with pending variants if one of those variants is already published. I think i'm going to stick with my original message and return all matched content regardless of published/trashed status